src/plugins/agent_api.go

```diff
@@ -605,6 +653,94 @@ func (h *NginxHandler) syncProcessInfo(processInfo []*core.Process) {
 	h.processes = processInfo
 }
 
+func (rh *RootHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set(contentTypeHeader, jsonMimeType)
+
+	switch {
+	case healthRegex.MatchString(r.URL.Path):
+		if r.Method != http.MethodGet {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		err := rh.healthCheck(w)
+		if err != nil {
+			log.Warnf("Failed to get agent health: %v", err)
+		}
+	default:
+		w.WriteHeader(http.StatusNotFound)
+		_, err := fmt.Fprint(w, []byte("not found"))
+		if err != nil {
+			log.Warnf("Failed to send api response: %v", err)
+		}
+	}
+}
+
+// swagger:route GET /health nginx-agent health-check
+//
+// # Check the health of the NGINX Agent
+//
+// responses:
+//
+//	200: HealthResponse
+func (rh *RootHandler) healthCheck(w http.ResponseWriter) error {
+	w.WriteHeader(http.StatusOK)
+
+	overallStatus := okStatus
+	checks := []HealthStatusCheck{}
+
+	registrationStatus := okStatus
+	commandServiceStatus := okStatus
+	metricsServiceStatus := okStatus
+
+	if rh.config.IsGrpcServerConfigured() {
+		if !rh.isGrpcRegistered {
+			registrationStatus = errorStatus
+			overallStatus = errorStatus
```

Comment:
Some of these could potentially be simplified to one line.
```suggestion
			registrationStatus, overallStatus = errorStatus, errorStatus
```